shared/modules/gas.utils.js

```diff
@@ -0,0 +1,99 @@
+} from './conversion.utils';
+
+/**
+ * Accepts an options bag containing gas fee parameters in hex format and
+ * returns a gasTotal parameter representing the maximum amount of wei the
+ * transaction will cost.
+ *
+ * @param {object} options - gas fee parameters object
+ * @param {string} [options.gasLimit] - the maximum amount of gas to allow this
+ *  transaction to consume. Value is a hex string
+ * @param {string} [options.gasPrice] - The fee in wei to pay per gas used.
+ *  gasPrice is only set on Legacy type transactions. Value is hex string
+ * @param {string} [options.maxFeePerGas] - The maximum fee in wei to pay per
+ *  gas used. maxFeePerGas is introduced in EIP 1559 and represents the max
+ *  total a user will pay per gas. Actual cost is determined by baseFeePerGas
+ *  on the block + maxPriorityFeePerGas. Value is hex string
+ * @returns {string} - The maximum total cost of transaction in hex wei string
+ */
+export function getMaximumGasTotalInHexWei({
+  gasLimit = '0x0',
+  gasPrice = '0x0',
+  maxFeePerGas,
+} = {}) {
+  if (maxFeePerGas) {
+    return addHexPrefix(
+      multiplyCurrencies(gasLimit, maxFeePerGas, {
+        toNumericBase: 'hex',
+        multiplicandBase: 16,
+        multiplierBase: 16,
+      }),
+    );
+  }
+  return addHexPrefix(
+    multiplyCurrencies(gasLimit, gasPrice, {
+      toNumericBase: 'hex',
+      multiplicandBase: 16,
+      multiplierBase: 16,
+    }),
+  );
+}
+
+/**
+ * Accepts an options bag containing gas fee parameters in hex format and
+ * returns a gasTotal parameter representing the minimum amount of wei the
+ * transaction will cost. For gasPrice types this is the same as max.
+ *
+ * @param {object} options - gas fee parameters object
+ * @param {string} [options.gasLimit] - the maximum amount of gas to allow this
+ *  transaction to consume. Value is a hex string
+ * @param {string} [options.gasPrice] - The fee in wei to pay per gas used.
+ *  gasPrice is only set on Legacy type transactions. Value is hex string
+ * @param {string} [options.maxFeePerGas] - The maximum fee in wei to pay per
+ *  gas used. maxFeePerGas is introduced in EIP 1559 and represents the max
+ *  total a user will pay per gas. Actual cost is determined by baseFeePerGas
+ *  on the block + maxPriorityFeePerGas. Value is hex string
+ * @param {string} [options.maxPriorityFeePerGas] - The maximum fee in wei to
+ *  pay a miner to include this transaction.
+ * @param {string} [options.baseFeePerGas] - The estimated block baseFeePerGas
+ *  that will be burned. Introduced in EIP 1559. Value in hex wei.
+ * @returns {string} - The minimum total cost of transaction in hex wei string
+ */
+export function getMinimumGasTotalInHexWei({
+  gasLimit = '0x0',
+  gasPrice,
+  maxPriorityFeePerGas,
+  maxFeePerGas,
+  baseFeePerGas,
+} = {}) {
+  if (gasPrice) {
+    return getMaximumGasTotalInHexWei({ gasLimit, gasPrice });
+  }
+  const minimumFeePerGas = addCurrencies(baseFeePerGas, maxPriorityFeePerGas, {
```

Comment:
Can't understand why `minimumFeePerGas` = `baseFeePerGas` + `maxPriorityFeePerGas` 🤔

Reply:
Here is where you can see background info on EIP 1559: https://hackmd.io/@q8X_WM2nTfu6nuvAzqXiTQ/1559-wallets

As for the `minimumFeePerGas`: we have to present a range of possible fees to the user. The lowest they could possibly pay is the `baseFee + maxPriorityFeePerGas`. This is because the `baseFee` is required and will be burned, and we can expect all of the `maxPriorityFeePerGas` to be taken by the miner. So, at a minimum, `baseFee + maxPriorityFeePerGas` will be used to mine/process the transaction